web/src/message_events.ts

```diff
@@ -35,19 +38,20 @@ import {realm} from "./state_data.ts";
 import * as stream_list from "./stream_list.ts";
 import * as stream_topic_history from "./stream_topic_history.ts";
 import * as sub_store from "./sub_store.ts";
+import type {UpdateMessageEvent} from "./types.ts";
 import * as unread from "./unread.ts";
 import * as unread_ui from "./unread_ui.ts";
 import * as util from "./util.ts";
 
-function filter_has_term_type(filter, term_type) {
+function filter_has_term_type(filter: Filter | undefined, term_type: string): boolean {
     return (
         filter !== undefined &&
```

Comment:
Accepting `Filter | undefined` here has no advantage; every caller either passes a defined `Filter` or has to later check for `undefined` anyway.

```suggestion
function filter_has_term_type(filter: Filter, term_type: string): boolean {
    return (
```